sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/implementation/OkHttpResponse.java

```diff
@@ -26,17 +25,21 @@ public OkHttpResponse(okhttp3.Response response, HttpRequest request, boolean ea
             eagerlyConvertHeaders
                 ? fromOkHttpHeaders(response.headers())
                 : new OkHttpToCoreHttpHeadersWrapper(response.headers()),
-            bodyBytes == null
-                ? response.body() == null
-                    ? EMPTY_BODY
-                    : BinaryData.fromStream(response.body().byteStream())
-                : BinaryData.fromBytes(bodyBytes));
+            null);
         // innerResponse.body() getter will not return null for server returned responses.
         // It can be null:
         // [a]. if response is built manually with null body (e.g. for mocking)
         // [b]. for the cases described here
         // [ref](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-response/body/).
         this.responseBody = response.body();
+
+        if (bodyBytes == null) {
+            if (responseBody != null && responseBody.contentLength() != 0) {
+                this.body = BinaryData.fromStream(responseBody.byteStream());
+            }
+        } else {
+            this.body = BinaryData.fromBytes(bodyBytes);
```

Comment:
Wouldn't this give an NPE because `bodyBytes`  is null?

Reply:
The else part of this check is for when `bodyBytes` is not `null`